src/hist/basehist.py

```diff
@@ -468,38 +459,57 @@ def plot2d(
 
     def plot2d_full(
         self,
+        *,
         fig: Optional[matplotlib.figure.Figure] = None,
         main_ax: Optional[matplotlib.axes.Axes] = None,
         top_ax: Optional[matplotlib.axes.Axes] = None,
         side_ax: Optional[matplotlib.axes.Axes] = None,
```

Comment:
I thought this was going to be `ax_dict =`

Reply:
I have implemented in this way hours ago. But abandoned, think about this:

```python
plot2d_full(fig=fig, ax_dict={"main": main_ax, "top": top_ax, "side": side_ax}, ...)
plot2d_full(fig=fig, main_ax=main_ax, top_ax=top_ax, side_ax=side_ax, ...)
```

Which would be a better shortcut for HEPists? I think the second, as they won't require structured manipulation -- encapsulate to a dict (or list, set ...) -- and only directly pass objects in. But if you want them to be better-structured, it's also reasonable.

Reply:
The problem is that your second example doesn't tell me that I need all 3 at the same time. There's no way in Python's signatures to indicate that one keyword argument will require the others. While the dict clearly tells me it it needs several axes. We can even make a TypedDict and MyPy might help a user add the correct keys.

I don't expect users to often make & pass the dict - I expect 90% of users to leave it empty, and many of the remaining 10% can pass `fig=` and still have the axes created for them.

Reply:
Oh, just to note that, as for the manipulation for the pass-in arguments, we can handle them in both ways.

Reply:
You can always write:

```python
plot2d_full(fig=fig, ax_dict=dict(main_ax=main_ax, top_ax=top_ax, side_ax=side_ax), ...)
```

Reply:
Okay, will change back to it 😁

Reply:
Do we also need to give ax_dict to plot1d and plot2d (one axis plotting) to keep the uniform of API?

Reply:
No, plotting onto multiple axes is different than the (more common) plot to one axis.